### PR TITLE
OS-8400 pkgsrc-setup should be able to re-bootstrap

### PR DESCRIPTION
**Pull request by bahamat**

### CODE REVIEW

**Comment by danmcd** [comment]
1.) There's no man page or usage information you need to update with this new `--clean` flag? (Blocking +1 for review.)
2.) Test notes in the ticket?  (Blocking IA)

File: src/smartdc/bin/pkgsrc-setup.sh [hunk 1]
```diff
@@ -18,6 +18,21 @@ if [[ -n "$TRACE" ]]; then
     set -o xtrace
 fi
 
+# If there's an existing install...
+if [[ -e /opt/tools/bin/pkgin ]]; then
+    # Re-bootstrap pkgsrc if requested.
+    if [[ $1 == '--clean' ]]; then
+        pkg_stash=$(mktemp)
+        pkgin export > "$pkg_stash"
+        rm -rf /opt/tools
```
Comment (danmcd) [inline src/smartdc/bin/pkgsrc-setup.sh:27]: Remove the $pkg_stash file too?

**Review by travispaul — APPROVED** [review]

File: src/smartdc/bin/pkgsrc-setup.sh (hunk 1 above)
Comment (danmcd) [reply src/smartdc/bin/pkgsrc-setup.sh:27]: Ooops, my bad. Taken care of later.

**Comment by danmcd** [comment]
@travispaul put some test notes in the ticket.

**Comment by danmcd** [comment]
And I answered my own question about man pages or other docs.

**Review by danmcd — APPROVED** [review]

File: man/smartdc/man/man1/pkgsrc-setup.1.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# pkgsrc-setup(1) -- bootstrap pkgsrc-tools
+
+
+## SYNOPSIS
+
+`pkgsrc-setup [--clean]`
+
```
Comment (danmcd) [inline man/smartdc/man/man1/pkgsrc-setup.1.md:7]: There's no mention of /path/to/root as an option.  I take it that's deliberate and not an oversight?

Comment (bahamat) [reply man/smartdc/man/man1/pkgsrc-setup.1.md:7]: Added a man page

Comment (bahamat) [reply man/smartdc/man/man1/pkgsrc-setup.1.md:7]: Correct. The path to root is only intended to be used during smartos install when the zpool is mounted to an alternate location. If you just tried to install to an arbitrary location I don't even know if the installation would be usable. On a running system, any location other than `/opt/tools` is not supported.

**Review by danmcd — APPROVED** [review]
Choosing to keep the /path/to/root undocumented is okay.

**Review by travispaul — APPROVED** [review]

**Pull request merged**